### ISSUE TITLE
Flexbox in wizard step

### ISSUE THREAD

**Issue by water42** [madoar/ng2-archwizard#26]
In this image the wizard-step does not get set to 100% of the wizard (which is set to 100% of its container in my app). So "space-between" does not have the desired effect.

![image](https://user-images.githubusercontent.com/12549876/29569420-75b7e628-8721-11e7-8423-da0ab750c76a.png)

code:

```
<wizard style="height: 100%;">
    <wizard-step style="
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;"
                 title="Establish Focus Component">
      <app-establish-focus></app-establish-focus>
      <div>
        <hr>
        <button class="btn btn-primary"
                type="button"
                previousStep
                disabled>Back</button>
        <button class="btn btn-primary"
                type="button"
                nextStep>Continue</button>
      </div>
    </wizard-step>
  </wizard>
```

Basically I want the component to start directly under the navigation and the HR to be at the bottom of the page.

**Comment by water42** [comment]
FWIW by adding `height: 100%` to the wizard component styles, my issue is solved. Not sure if this is a bug or I was using flex boxes wrong.

**Comment by madoar** [comment]
I'm not sure I understand your request.
Basically you want to make the wizard fill out the full height of your browser window, is this correct?

I've never tried this before, but basically you would need to give the wizard a height of `100%` and then tell the wizard step to fill the available space completely. 
The last part is a little problematic, because the `wizard-step` component, which you're using defines its own css, which sets the height of the step to `auto`.

My guess is, that you'll need to create your use custom step components via the `wizardStep` directive, to freely set the height of the wizard step. A short example about how to define your own custom step components can be found at https://madoar.github.io/ng2-archwizard-demo/#/wizard-step-directives, whose source code can be found at https://github.com/madoar/ng2-archwizard-demo/tree/master/src/app/wizard-step-directives and https://github.com/madoar/ng2-archwizard-demo/tree/master/src/app/custom-step-component.

I'm not sure about the `ng-content` part of your last comment, could you please provide some additional information?

**Comment by water42** [comment]
Hm yeah unfortunately I don't entirely understand what's happening. Here is the element I need to have a height: 100%:

`<div _ngcontent-c4="" class="wizard-steps horizontal" ng-reflect-ng-class="[object Object]">` 

In my original screenshot, it is the element that is the parent of the selected wizard-step. If this element does not have height 100% (I'm setting it manually by interacting with the DOM) then having 100% on wizard steps is useless.

Does this make more sense?

**Comment by water42** [comment]
Custom wizard steps served as a good enough solution to the problem so I'll close this for now. thanks

**Comment by madoar** [comment]
The overall problem with a fullscreen mode for `ng2-archwizard` is, that it requires quite a few additional `height: 100%` settings, like you previously discovered. 
If the wizard shouldn't be shown in fullscreen, these height settings need to be removed, because otherwise some strange height effects can occur.
The only good solution I can imagine for this is to add an input to the `wizard` component, where you can specify if you want to use the wizard in fullscreen mode or not. 
This setting would then need to add the height setting to the css.

Would such an input solve your request?

**Comment by water42** [comment]
Yeah that would be great. Can I try implementing that feature?

**Comment by madoar** [comment]
Feel free to give it a go :)
I think I'll have some time next week to work on `ng2-archwizard` again.

**Comment by gmarab** [comment]
Any update on this? 
I would also need to use this wonderful component in full video mode.
Thanks!

**Comment by madoar** [comment]
Currently none.
@water42 do you still plan on implementing the feature?
Otherwise I plan to add this feature to the second next release.

**Comment by madoar** [comment]
Just to give you a short heads up:
I currently don't think we need to add a new input to the wizard component for this feature, because I believe this can also be achieved by using #27 and #46.

**Comment by madoar** [comment]
Closed by #92.
I believe the height of the wizard should be changeable by overwriting the wizard `height` css property, which is possible with the changed introduced by #92.
